### DISCUSSION TITLE
PEP484 compliance

### DISCUSSION THREAD

**Discussion by Naviss** [OpenGATE/opengate#310] [Technical issues]
I've encountered significant challenges with GATE and macro files in the past, primarily due to inadequate completion features and limited language support in contemporary IDEs. In my experience, self-documenting code becomes highly feasible when the IDE efficiently tracks definitions and offers predictive code completion.

With modern Python, this process is greatly facilitated and simplified by adhering to PEP 484, which pertains to type hinting. This standard greatly enhances the capability of IDEs to understand and interact with the code.

Given these considerations, I'm curious to know whether OpenGATE plans to incorporate these practices in the future. Will the project align with PEP 484 or similar standards to enhance IDE compatibility and code readability? urthermore, I'm interested in contributing to the project. Would contributions that align with these principles be welcome in OpenGATE?

**Comment by dsarrut** [comment]
Hi @Naviss, 

yes indeed, this is a good idea. We are already trying to have "some" integration for IDE. With the recent restructuration, it evolves in this direction and IDE should help with some code prediction and definition helpers (at least PyCharm, but I think it could be similar with VScode). It is, however, far from complete. Also, some user help is now included, allowing to print and discover users parameters. 

PEP484, is about code type hints right? It could be done, for sure, but I am unsure if it is worth it (and there are already a lot of other things to do). I am open for discussion here. 

And finally, yes! you are very welcome for any kind of contribution! Please contact me by email to setup a zoom meeting.
